config/common.go

```diff
@@ -110,6 +110,7 @@ func (cs *commonStore) load(f io.ReadCloser, needsSave bool, validate func(*mode
 	needsSave = needsSave || loadedCfg.FileSettings.PublicLinkSalt == nil || len(*loadedCfg.FileSettings.PublicLinkSalt) == 0
 
 	loadedCfg.SetDefaults()
+	loadedCfgWithoutEnvOverrides.SetDefaults()
```

Comment:
This line worries me slightly: `SetDefaults` isn't guaranteed to be deterministic. For example, `AtRestEncryptKey`, if empty, is generated at random. Is there a weird edge case where `loadedCfg` and `loadedCfgWithoutEnvOverrides` diverge on values that weren't actually overridden, and we take the "wrong" value?

Would it be possible to push the `nil` handling logic downstream in `removeEnvOverrides` and simply handle the case that `loadedCfgWithoutEnvOverrides` (later called `cfgWithoutEnv`) isn't fully defined? It's normal for a `model.Config` to have `nil` parameters -- e.g. a `PATCH` against config would look like this. I think it's similarly reasonable to model the loaded values from the config, and properly model that some values simply weren't there.